maporaclespatial.c

```diff
@@ -2104,6 +2143,35 @@ int msOracleSpatialLayerWhichShapes( layerObj *layer, rectObj rect, int isQuery)
   if (layer->debug)
     msDebug("msOracleSpatialLayerWhichShapes. Using this Sql to retrieve the data: %s\n", query_str);
 
+  /* db_role is a custom metadata. When this is set, that means we should use this role to access db. */
+  char *db_role = msLookupHashTable(&layer->metadata, "db_role");
+  if (db_role) {
+    if (layer->debug >= 3)
+      msDebug("msOracleSpatialLayerWhichShapes. Call Set_User with user: %s.\n", db_role);
+    success = setDBRole(hand, sthand, db_role);
+
+    /* If calling on Set_User fails, clean up and return. */
+    if (!success) {
+      /* Log message. */
+      msDebug("Error: %s . "
+          "Failed to call Set_User() on database."
+          "msOracleSpatialLayerWhichShapes()\n", hand->last_oci_error);
+      msSetError(MS_ORACLESPATIALERR,
+          "Check your data statement and server logs",
+          "msOracleSpatialLayerWhichShapes()");
+
+      /* Clean up. */
+      free(items);
+      if (geom_column_name) free(geom_column_name);
+      if (srid) free(srid);
+      if (unique) free(unique);
+      if (indexfield) free(indexfield);
+      free(table_name);
+      free(query_str);
+      return MS_FAILURE;
```

Comment:
Do we need to call resetDBRole here ?

Reply:
I'm not sure if it's necessary to call resetDBRole here. Because when this block of code is run, that means the previous setDBRole fails.
And inside function setDBRole, there are 3 function callings:
1. OCIStmtPrepare
2. OCIBindByName
3. OCIStmtExecute
Notice OCIStmtExecute is the last one that actually run the underlying stored procedure. In my opinion, any of these functions fail, calling the cleanup function resetDBRole is not very necessary. That being said, if you insist I can add this as you suggest.